gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -110,29 +108,27 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
           continue;
         }
 
-        if (this.fss.get(branchId).exists(publisherOutputDir)) {
-
-          // The final output directory already exists, check if the job is configured to replace it
-          boolean replaceFinalOutputDir =
-              this.getState().getPropAsBoolean(
-                  ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.DATA_PUBLISHER_REPLACE_FINAL_DIR,
-                      this.numBranches, branchId));
-
-          // If the final output directory is not configured to be replaced, then append the new data to the existing
-          // output folder
-          if (!replaceFinalOutputDir) {
-            addWriterOutputToExistingDir(writerOutputDir, publisherOutputDir, workUnitState, branchId);
-            writerOutputPathsMoved.add(writerOutputDir);
-            continue;
+        if (this.fss.get(branchId).exists(writerOutputDir)) {
```

Comment:
But then `workUnitState.setWorkingState(WorkUnitState.WorkingState.COMMITTED)` will be skipped.